### PR TITLE
Add R2 scopes for api_token_permission_groups

### LINKED ISSUE #2684: Support R2 scope for token permissions in `api_token_permission_groups` datasource
### Current Terraform and Cloudflare provider version

Terraform v1.5.0
on windows_amd64
+ provider registry.terraform.io/cloudflare/cloudflare v4.12.0

### Description

Now that R2 bucket scoped tokens are out, there is a new scope of `com.cloudflare.edge.r2.bucket` and it would be nice if it was a separate scope in the [api_token_permission_groups datasource](https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/internal/sdkv2provider/data_source_api_token_permission_groups.go#L68-L77) rather than having to use the legacy `permissions` output.

I can work on this if accepted and a name chosen, maybe just `r2`.

### Use cases

Making R2 API tokens that are bucket scoped.

### Potential Terraform configuration

```hcl
data "cloudflare_api_token_permission_groups" "all" {}

# Get R2 Write permssions.
output "r2_write_permissions" {
  value = data.cloudflare_api_token_permission_groups.all.r2["Workers R2 Storage Bucket Item Write"] 
}
```

### References

Part of a custom module I wrote
https://github.com/Cyb3r-Jak3/terraform-cloudflare-api-permissions/blob/main/outputs.tf#L13-L16
Permissions
https://api.cyberjake.xyz/cloudflare_api/token_permissions?scope=r2

### PR DESCRIPTION
**Pull request by Cyb3r-Jak3**
Adds support for new R2 permission scopes. Also converts to the plugin framework
Closes #2684

### CODE REVIEW

**Comment by Cyb3r-Jak3** [comment]
Marking as draft because I'm getting a weird issue of import cycle with
```bash
root@de78b1a86397:/workspaces/terraform-provider-cloudflare# TF_ACC=1 go test ./internal/framework/service/api_token_permissions_groups -v -count 1 -run ^TestAccCloudflareApiTokenPermissio
nGroups_Basic -timeout 120m -parallel 1
# github.com/cloudflare/terraform-provider-cloudflare/internal/framework/service/api_token_permissions_groups
package github.com/cloudflare/terraform-provider-cloudflare/internal/framework/service/api_token_permissions_groups
        imports github.com/cloudflare/terraform-provider-cloudflare/internal/framework/provider
        imports github.com/cloudflare/terraform-provider-cloudflare/internal/acctest: import cycle not allowed in test
FAIL    github.com/cloudflare/terraform-provider-cloudflare/internal/framework/service/api_token_permissions_groups [setup failed]
```
Not entirely sure why that is because the rulesets resource_test has the same imports

File: internal/framework/service/api_token_permissions_groups/data_source_test.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
+package api_token_permissions_groups
```
Comment (jacobbednarz) [inline internal/framework/service/api_token_permissions_groups/data_source_test.go:1]: this is why you are getting cyclic import issues. the test packages are suffixed with `_test`

```suggestion
package api_token_permissions_groups_test
```

**Comment by jacobbednarz** [comment]
this is a great initiative but i'm also happy if you just do this in the SDKv2 instead. we don't need to solve all of this at once :)

File: internal/framework/service/api_token_permissions_groups/data_source.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+}
+
+// stringChecksum takes a string and returns the checksum of the string.
+func stringChecksum(s string) string {
+	h := md5.New()
+	h.Write([]byte(s))
+	bs := h.Sum(nil)
+
+	return fmt.Sprintf("%x", bs)
+}
```
Comment (Cyb3r-Jak3) [inline internal/framework/service/api_token_permissions_groups/data_source.go:117]: This should probably be moved somewhere else, but not sure where the best place would be.

File: internal/framework/service/api_token_permissions_groups/data_source_test.go (hunk 1 above)
Comment (Cyb3r-Jak3) [reply internal/framework/service/api_token_permissions_groups/data_source_test.go]: Thanks!

**Comment by Cyb3r-Jak3** [comment]
I've removed the docs changes that aren't related to this change to prevent merge issues.

**Comment by jacobbednarz** [comment]
acceptance tests all passing

```
TF_ACC=1 go test ./internal/framework/service/api_token_permissions_groups -v -run "^TestAccCloudflareApiTokenPermissionGroups_" -count 1 -timeout 120m -parallel 1
=== RUN   TestAccCloudflareApiTokenPermissionGroups_Basic
=== PAUSE TestAccCloudflareApiTokenPermissionGroups_Basic
=== CONT  TestAccCloudflareApiTokenPermissionGroups_Basic
--- PASS: TestAccCloudflareApiTokenPermissionGroups_Basic (13.68s)
PASS
ok  	github.com/cloudflare/terraform-provider-cloudflare/internal/framework/service/api_token_permissions_groups	14.582s
```

**Pull request merged**